spark/src/main/scala/org/apache/comet/CometExecIterator.scala

```diff
@@ -63,9 +64,28 @@ class CometExecIterator(
   }.toArray
   private val plan = {
     val conf = SparkEnv.get.conf
-    // Only enable unified memory manager when off-heap mode is enabled. Otherwise,
-    // we'll use the built-in memory pool from DF, and initializes with `memory_limit`
-    // and `memory_fraction` below.
+
+    val offHeapMode = CometSparkSessionExtensions.isOffHeapEnabled(conf)
+    val memoryLimit = if (offHeapMode) {
+      // in unified mode we share off-heap memory with Spark
```

Comment:
I noticed that not all memory allocations in Datafusion grow the memory_pool, if we share all off-heap memory to native, will it cause out off memory?